### PR TITLE
Feature/kotlin refactoring

### PR DESCRIPTION
**Pull request by jankowskib**
###### Addresses issue #.
- [x] This pull request follows the coding standards

###### This PR changes:
 - Created extensions for a few classes

### CODE REVIEW

File: library/src/main/java/com/devbrackets/android/exomedia/util/DeviceUtil.kt [hunk 1]
```diff
@@ -20,44 +20,18 @@ import android.app.UiModeManager
...
-        protected val NON_COMPATIBLE_DEVICES: MutableList<Device>
 
-        init {
-            NON_COMPATIBLE_DEVICES = LinkedList()
-            NON_COMPATIBLE_DEVICES.add(Device("Amazon"))
-        }
-    }
+    private val NON_COMPATIBLE_DEVICES = listOf(Device("Amazon"))
 
     fun supportsExoPlayer(context: Context): Boolean {
```
Comment (brianwernick) [inline library/src/main/java/com/devbrackets/android/exomedia/util/DeviceUtil.kt]: Long run we should make the `DeviceUtil` and interface and this the default implementation to help with customization (was a non-final class in Java to do this). I don't expect that to be done in this PR but I do want to make sure we do it before 5.0 is released

File: library/src/main/java/com/devbrackets/android/exomedia/util/DeviceUtil.kt [hunk 2]
```diff
@@ -20,44 +20,18 @@ import android.app.UiModeManager
...
-        return if (!isNotCompatible(NON_COMPATIBLE_DEVICES)) {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN && !isNotCompatible()) {
```
[the remark this answers is not in the document]
Comment (jankowskib) [reply library/src/main/java/com/devbrackets/android/exomedia/util/DeviceUtil.kt]: Yeah right! I fixed this.

File: library/src/main/java/com/devbrackets/android/exomedia/util/DeviceUtil.kt (hunk 1 above)
Comment (jankowskib) [reply library/src/main/java/com/devbrackets/android/exomedia/util/DeviceUtil.kt]: How would the interface look like? Could it be something like this?
``interface DeviceUtil {
    fun supportsExoPlayer(context: Context): Boolean
}
``

Comment (brianwernick) [reply library/src/main/java/com/devbrackets/android/exomedia/util/DeviceUtil.kt]: Pretty much though I would probably remove the context as an arg and let that be an implementation detail

**Pull request merged**